pkg/cloud/vsphere/provisioner/govmomi/delete.go

```diff
@@ -20,52 +21,68 @@ func (pv *Provisioner) Delete(ctx context.Context, cluster *clusterv1.Cluster, m
 	if cluster == nil {
 		return errors.New(constants.ClusterIsNullErr)
 	}
+	if exists, _ := pv.Exists(ctx, cluster, machine); exists {
+		if err := pv.powerOffAndDestroy(ctx, cluster, machine); err != nil {
+			return err
```

Comment:
i think we should start to return `controllerError.RequeueAfterError` in order for controller to re-queue, otherwise, there will be many cases that machine object been deleted, but those VMs or nodes never been deleted.